pkg/controller/operators/catalog/subscription/reconciler.go

```diff
@@ -91,11 +91,16 @@ func (c *catalogHealthReconciler) Reconcile(ctx context.Context, in kubestate.St
 
 				var healthUpdated, deprecationUpdated bool
 				next, healthUpdated = s.UpdateHealth(c.now(), catalogHealth...)
+				if healthUpdated {
+					if _, err := c.client.OperatorsV1alpha1().Subscriptions(ns).UpdateStatus(ctx, s.Subscription(), metav1.UpdateOptions{}); err != nil {
+						return nil, err
+					}
+				}
 				deprecationUpdated, err = c.updateDeprecatedStatus(ctx, s.Subscription())
 				if err != nil {
 					return next, err
 				}
-				if healthUpdated || deprecationUpdated {
+				if deprecationUpdated {
```

Comment:
So, this case we don't check for error... but the health case we do now?

Reply:
Daniel and I had a conversation about this here https://github.com/operator-framework/operator-lifecycle-manager/pull/3349#discussion_r1705814505

Reply:
If you mean to say that this [error](https://github.com/operator-framework/operator-lifecycle-manager/pull/3349/files#diff-1452f55a2f3e0dd96fcb5d63a2a0de2244790bcd067225f4989044330006803aL99) is unchecked, I noticed it too, but left it alone since it isn't part of the PR. But since I'm in this area I could totally just add that in if that's what you were asking for.

Reply:
It's probably not clear because the diff cuts off the func definition, but the error is unchecked because this func uses named returns, and this is the last potential error-causing statement, so that error will be returned if there is one. I'm not defending it, I really don't like named returns, just explaining why it looks like that.

Reply:
Ah no that's sounds fine to me. It's one of those "there's 2 ways of doing this and both are right", but it's not related to this PR.